versioned/gc/iceberg/pom.xml

```diff
@@ -30,11 +30,13 @@
 
   <name>Nessie - Versioned - GC for Spark - Iceberg</name>
 
-  <properties>
-    <iceberg.nessie.version>${project.version}</iceberg.nessie.version>
-  </properties>
-
   <dependencies>
+    <dependency>
+      <groupId>org.projectnessie</groupId>
+      <artifactId>nessie-client</artifactId>
```

Comment:
for this and above I would rather not add a nessie version explicitly. There is one in the iceberg runtime and we want to make sure the iceberg runtime one is used. As taht is how it will be used in production. This could mask issues w/ the client versions

Reply:
let me double-check whether this works without adding an explicit dependency to `nessie-client`

Reply:
ah now I remember why it was failing. I missed the fact that `iceberg.nessie.version` was defined in each of the GC poms and I only updated one of them and wondered why it failed, thus leading me to adding an explicit `nessie-client` dependency

Reply:
fixed it

Reply:
ok i double checked this and it doesn't work without forcing the `nessie-client` version here to be `${iceberg.nessie.version}`. It's either that, or we run into the jackson issue that you already ran into (where you moved the `iceberg-spark3-runtime` dependency to the very end of the pom:
```
Caused by: org.apache.iceberg.shaded.com.fasterxml.jackson.databind.exc.InvalidDefinitionException: 
Cannot construct instance of `org.projectnessie.model.Reference` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
```